src/app/packages/menu/menu/menu.component.ts

```diff
@@ -53,22 +52,23 @@ export class MenuComponent implements AfterViewInit {
 	@ViewChild('container') container: ElementRef;
 	@ViewChild('selected_component_elem', { read: ViewContainerRef }) selected_component_elem: ViewContainerRef;
 
-	menu_items$: Observable<MenuItem[]> = this.store.select('menu')
-		.map((state: IMenuState) => state.menu_items)
+	menuState$: Observable<IMenuState> = this.store.select('menu');
+
+	menu_items$: Observable<MenuItem[]> = this.menuState$
+		.pluck <IMenuState, MenuItem[]> ('menu_items')
 		.distinctUntilChanged(isEqual);
 
 	menu_items: MenuItem[];
 
-	selected_menu_item_index$: Observable<number> =
-		this.store.select('menu')
-			.map((state: IMenuState) => state.selected_menu_item_index)
-			.distinctUntilChanged(isEqual);
+	selected_menu_item_index$: Observable<number> = this.menuState$
+			.pluck <IMenuState, number> ('selected_menu_item_index')
```

Comment:
you can keep this as is, I will fix it later with a tslint parameter (the spaces for typing)